File: Day24/addStrings.cpp
class Solution
{
public:
    string addStrings(string num1, string num2)
    {
        int n1 = num1.size() - 1;
        int n2 = num2.size() - 1;

        string s = "";
        int carry = 0;

        while (n1 >= 0 || n2 >= 0 || carry >= 1)
        {

            int digit1 = n1 >= 0 ? num1[n1--] - '0' : 0; //- '0' converst. char to int
            int digit2 = n2 >= 0 ? num2[n2--] - '0' : 0;

            int sum = digit1 + digit2 + carry;
            cout << sum << endl;
            s = to_string(sum % 10) + s;
            carry = sum / 10;
        }
        return s == "" ? "0" : s;
    }
};

File: Day24/AddStrings.java
class Solution {
    public String addStrings(String num1, String num2) {
        if(num1 == null) return num2;
        if(num2 == null) return num1;
        
        StringBuilder sb = new StringBuilder();
        int i = num1.length()-1;
        int j = num2.length()-1;
        
        int carry  = 0;
        
        while(i >= 0 || j >= 0){
            int a = i >= 0 ? num1.charAt(i--)-'0' : 0; 
            int b = j >= 0 ? num2.charAt(j--)-'0' : 0;
            int sum = a + b + carry;
            
            carry  = sum / 10;
            sb.append(sum % 10);
        }
        
        if(carry != 0) sb.append(carry);
        return sb.reverse().toString();
    }
}
